src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -93,7 +93,10 @@ public static class PrereleaseCompatibilityTransformer
                 // above this call). We are required to regenerate it, however, in order to properly 
                 // elide default values, etc. I could not find a way for the JToken driven
                 // ToString()/text-generating mechanism to honor default value ignore/populate settings.
-                updatedLog = JsonConvert.SerializeObject(transformedSarifLog);
+                if (forceUpdate || modifiedLog)
```

Comment:
>forceUpdate [](start = 20, length = 11)

real issue here is that we weren't honoring the formatting param. added the forceUpdate/modifiedLog checks just to minimize code execution for certain cases